src/linalg.jl

```diff
@@ -229,14 +229,20 @@ end
         return :(_init_zero(a))
     end
 
-    expr = :(LinearAlgebra.norm_sqr(a[1]))
+    expr = :(LinearAlgebra.norm_sqr(a[1]/aₘ))
     for j = 2:prod(S)
-        expr = :($expr + LinearAlgebra.norm_sqr(a[$j]))
+        expr = :($expr + LinearAlgebra.norm_sqr(a[$j]/aₘ))
     end
 
     return quote
         $(Expr(:meta, :inline))
-        @inbounds return sqrt($expr)
+        zero_a = _init_zero(a)
+        aₘ = mapreduce(norm, max, a)::typeof(zero_a)
```

Comment:
The return type assertion here is to avoid type instability when `a` is a vector of vectors (https://github.com/JuliaArrays/StaticArrays.jl/blob/master/test/linalg.jl#L278).  `mapreduce` produces a type-unstable output for such `a`.  If there is a better approach without using return type assertion, please make a suggestion.